plugin/storage/grpc/options.go

```diff
@@ -29,10 +29,12 @@ const (
 	pluginBinary             = "grpc-storage-plugin.binary"
 	pluginConfigurationFile  = "grpc-storage-plugin.configuration-file"
 	pluginLogLevel           = "grpc-storage-plugin.log-level"
+	pluginWriterType         = "grpc-storage-plugin.writer-type"
```

Comment:
I think we had a mis-communication. I don't know if this flag is needed, because if the plugin implements streaming API I don't see why it shouldn't be used automatically, can you think of any reasons?  Even if there are reasons, the plugin author is in control of the Capabilities response and could implement their own configuration to disable streaming if there users want that.

When I was talking about a cmd line option I was only referring to the mock plugin we use in integration tests, where automatic upgrade to streaming API meant the regular API would never be tested.

Granted, this version achieves the same more generically, I am not too opposed to it, but it makes the streaming feature a bit more clunky to use imo. Perhaps we could design it so that the streaming upgrade still happens automatically, but the option allows to turn it off. For example, we could make the default value for this flag to be `auto` instead of `unary`.

Reply:
I also think this flag may not be necessary.
Now the plugin only accepts the `-config` flag, so I'm not sure how to use a command-line option to instruct the plugin whether to fill in the streamWriter field, maybe add a field to the memory plugin's config for the integration test as my latest commit, or do you think there is a more appropriate way?